seacatauth/external_login/providers/generic.py

```diff
@@ -260,6 +267,15 @@ def _get_verified_claims(self, id_token, expected_nonce: str | None = None):
 		except jwcrypto.jwt.JWTExpired:
 			L.error("Expired ID token.", struct_data={"provider": self.Type})
 			raise ExternalOAuthFlowError("Expired ID token.")
+		except jwcrypto.jwt.JWTMissingKey:
+			L.error("Error reading ID token - Invalid key in JWKSet.", struct_data={
+				"provider": self.Type})
+			# provider probably change jwks - refresh them
+			await self._prepare_jwks()
```

Comment:
I suggest to do this also during housekeeping.
Please see ASAB PubSub -> https://docs.teskalabs.com/asab/reference/pubsub/#well-known-messages

`Application.housekeeping!`